x-pack/plugins/inference/server/chat_complete/adapters/gemini/process_vertex_stream.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Observable } from 'rxjs';
+import {
+  ChatCompletionChunkEvent,
+  ChatCompletionTokenCountEvent,
+  ChatCompletionEventType,
+} from '../../../../common/chat_complete';
+import type { GenerateContentResponseChunk } from './types';
+
+export function processVertexStream() {
+  return (source: Observable<GenerateContentResponseChunk>) =>
+    new Observable<ChatCompletionChunkEvent | ChatCompletionTokenCountEvent>((subscriber) => {
+      function handleNext(value: GenerateContentResponseChunk) {
+        // completion: only present on last chunk
+        if (value.usageMetadata) {
+          subscriber.next({
+            type: ChatCompletionEventType.ChatCompletionTokenCount,
+            tokens: {
+              prompt: value.usageMetadata.promptTokenCount,
+              completion: value.usageMetadata.candidatesTokenCount,
+              total: value.usageMetadata.totalTokenCount,
+            },
+          });
+        }
+
+        const contentPart = value.candidates![0].content.parts[0];
+        const completion = contentPart.text;
+        const toolCall = contentPart.functionCall;
+
+        if (completion || toolCall) {
+          subscriber.next({
+            type: ChatCompletionEventType.ChatCompletionChunk,
+            content: completion ?? '',
+            tool_calls: toolCall
+              ? [
+                  {
+                    index: 0,
+                    toolCallId: '',
```

Comment:
I assume this means there is never a tool call id, and Gemini doesn't support parallel function calling at all? If that is the case, can we use `generateFakeToolCallId` from https://github.com/elastic/kibana/pull/190433? (Would be better than an empty string I think)

Reply:
> there is never a tool call id

Correct

> Gemini doesn't support parallel function calling at all

That's my understanding, and [this](https://www.googlecloudcommunity.com/gc/AI-ML/Gemini-1-5-parallel-function-calls/m-p/769179) seems to confirm it.

> can we use generateFakeToolCallId from #190433? (Would be better than an empty string I think)

Yeah, I guess that would make sense, will adapt

Reply:
done in https://github.com/elastic/kibana/pull/191292/commits/b5145d579d37f9ae187c9bc61e37db426b2f24f7